api/keepalive/index.ts

```diff
@@ -0,0 +1,27 @@
+import axios from 'axios'
+import * as express from 'express'
+import {app} from '../application'
+import {config} from '../config'
+
+const cookieToken = config.cookies.token
+
+async function handleAddressRoute(req, res) {
+  try {
+    const tokenSet = await app.locals.client.refresh(req.session.passport.user.tokenset.refresh_token)
```

Comment:
Probably need a check to see req.session.passport.user.tokenset.refresh_token is available. If not send 400 Bad request

Reply:
@ronaldmansveld this particular endpoint is existent for this sole purpose, it only gets called as per the keepalive (which is currently set to 5hrs). Agreed we do also need to perhaps do some additional checking for bearer access_token expiry upon an incoming request (this was raised but agreed that the 'heartbeat/keepalive' from frontend would mitigate this).

Reply:
Hang on, if we arent validating the access token expiry, what are we updating the front end with as this not persisted in our store to check unless ui will call the backend to validate this?

Reply:
we know that the access token has life span of 8hrs... so every 5 hrs is user is in not in idle state we are extending that session for another 8 hrs. we could extend this to get the actual timestamp and only then extend... but i think we are lucky that we got this far considering the history of this story

Reply:
I do see issues with the fact that we're 'hardcoding' values of 5 and 8 hours. The moment IDAM changes the lifetime of tokens etc, this will all come crashing down.
I appreciate if we decide to keep this for now due to time-constraints, but when this moves to the node common lib it really should only use data coming from the token itself, instead of hardcoded values.